sql/rpl_gtid.cc

```diff
@@ -3361,13 +3363,26 @@ static my_bool report_audit_findings(void *entry, void *report_ctx_arg)
   return FALSE;
 }
 
-my_bool Binlog_gtid_state_validator::report(FILE *out, my_bool is_strict_mode)
+my_bool Binlog_gtid_state_validator::report(FILE *out, my_bool is_strict_mode,
+                                            my_bool report_out_of_order,
+                                            rpl_gtid *stop_gtids, size_t n_stop_gtids)
 {
-  struct gtid_report_ctx report_ctx;
-  report_ctx.out_file= out;
-  report_ctx.is_strict_mode= is_strict_mode;
-  report_ctx.contains_err= FALSE;
+  struct gtid_report_ctx report_ctx= {out, is_strict_mode, report_out_of_order, FALSE};
   my_hash_iterate(&m_audit_elem_domain_lookup, report_audit_findings, &report_ctx);
+  for (size_t i= 0; i < n_stop_gtids; ++i)
+  {
+    rpl_gtid *stop_gtid= &stop_gtids[i];
+    struct audit_elem *audit_elem= (struct audit_elem *)my_hash_search(
+      &m_audit_elem_domain_lookup,
+      (const uchar *)&(stop_gtid->domain_id), 0
+    );
+    if (!(audit_elem && audit_elem->last_gtid.seq_no >= stop_gtid->seq_no)) // `>` case should not occur
+      Binlog_gtid_state_validator::warn(
+        out,
+        "Did not reach stop position %u-%u-%llu before end of input",
```

Comment:
I'd say to add a comment where MDEV-27037 / #3400 added its warnings, to mention that when `--stop-position` is provided a GTID, that a warning is also emitted (and where/how)

Reply:
Hello, please review my code comments.